plugins/woocommerce-blocks/tests/e2e/tests/product-collection/product-collection.block_theme.spec.ts

```diff
@@ -961,3 +968,253 @@ test.describe( 'Product Collection', () => {
+			collection:
+				'woocommerce/product-collection/my-custom-collection-product-context',
+		},
+		myCustomCollectionMultipleContexts: {
+			name: 'My Custom Collection - Multiple Contexts',
+			label: 'Block: My Custom Collection - Multiple Contexts',
+			collection:
+				'woocommerce/product-collection/my-custom-collection-multiple-contexts',
+		},
+	};
+
+	// Activate plugin which registers custom product collections
+	test.beforeEach( async ( { requestUtils } ) => {
+		await requestUtils.activatePlugin(
+			'register-product-collection-tester'
+		);
+	} );
+
+	Object.entries( MY_REGISTERED_COLLECTIONS_THAT_NEEDS_PRODUCT ).forEach(
+		( [ key, collection ] ) => {
+			test( `For collection "${ collection.name }" - manually selected product reference should be available on Frontend in a post`, async ( {
+				pageObject,
+				admin,
+				page,
+			} ) => {
+				await admin.createNewPost();
+				await pageObject.insertProductCollection();
+				await pageObject.chooseCollectionInPost( key as Collections );
+
+				// Verify that product picker is shown in Editor
+				const editorProductPicker = admin.page.locator(
+					SELECTORS.productPicker
+				);
+				await expect( editorProductPicker ).toBeVisible();
+
+				// Once a product is selected, the product picker should be hidden
+				await pageObject.chooseProductInEditorProductPickerIfAvailable(
+					admin.page
+				);
+				await expect( editorProductPicker ).toBeHidden();
+
+				// On Frontend, verify that product reference is a number
+				await pageObject.publishAndGoToFrontend();
+				const collectionWithProductContext = page.locator(
+					`[data-collection="${ collection.collection }"]`
+				);
+				const queryAttribute = JSON.parse(
+					( await collectionWithProductContext.getAttribute(
+						'data-query'
+					) ) || '{}'
+				);
+				expect( typeof queryAttribute?.productReference ).toBe(
+					'number'
+				);
+			} );
+
+			test( `For collection "${ collection.name }" - product picker shouldn't be shown in Single Product template`, async ( {
+				pageObject,
+				admin,
+				editor,
+			} ) => {
+				await admin.visitSiteEditor( {
+					postId: `woocommerce/woocommerce//single-product`,
+					postType: 'wp_template',
+					canvas: 'edit',
+				} );
+				await editor.canvas.locator( 'body' ).click();
+				await pageObject.insertProductCollection();
+				await pageObject.chooseCollectionInTemplate(
+					key as Collections
+				);
+
+				const editorProductPicker = editor.canvas.locator(
+					SELECTORS.productPicker
+				);
+				await expect( editorProductPicker ).toBeHidden();
+			} );
+		}
+	);
+
+	test( 'Product picker should work as expected while changing collection using "Choose collection" button from Toolbar', async ( {
+		pageObject,
+		admin,
+	} ) => {
+		await admin.createNewPost();
+		await pageObject.insertProductCollection();
+		await pageObject.chooseCollectionInPost(
+			'myCustomCollectionWithProductContext'
+		);
```

Comment:
Ditto, can be replaced with:

```suggestion
		await pageObject.createNewPostAndInsertBlock( 'myCustomCollectionWithProductContext' );
```

Reply:
As I explained [here](https://github.com/woocommerce/woocommerce/pull/50590#discussion_r1738359872), I can't use it because `createNewPostAndInsertBlock` also select a product by calling `chooseProductInEditorProductPickerIfAvailable` function. For this test, I wanna call it manually. Does it make sense?